File: MAD Lab6/app/src/main/java/com/sliit/madlab6/MainActivity.java
package com.sliit.madlab6;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button ok = findViewById(R.id.btn_ok);
        EditText readerName = findViewById(R.id.editTextName);

        ok.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(MainActivity.this,ReadingCollectionsActivity.class);
                String reader = readerName.getText().toString().trim();
                intent.putExtra("readerName",reader);
                startActivity(intent);
            }
        });
    }
}

File: MAD Tutorial7/app/src/main/java/com/sliit/madtutorial7/MainActivity.kt
package com.sliit.madtutorial7

import android.icu.util.Calendar
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val inputYear = findViewById<EditText>(R.id.input_year)
        val txtAge = findViewById<TextView>(R.id.txtAge)
        val btn_submit = findViewById<Button>(R.id.btn_submit)

        btn_submit.setOnClickListener{
            val currentYear = Calendar.getInstance().get(Calendar.YEAR)
            txtAge.text = "Age : " + (currentYear - inputYear.text.toString().trim().toInt())
        }

        /*fun onSubmit(view: View) {
        }*/
    }


}